src/libraries/System.Net.Http.Json/ref/System.Net.Http.Json.cs

```diff
@@ -0,0 +1,43 @@
+        public static System.Threading.Tasks.Task<System.Net.Http.HttpResponseMessage> PostAsJsonAsync(this System.Net.Http.HttpClient client, string requestUri, System.Type type, object value, System.Text.Json.JsonSerializerOptions options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static System.Threading.Tasks.Task<System.Net.Http.HttpResponseMessage> PostAsJsonAsync(this System.Net.Http.HttpClient client, System.Uri requestUri, System.Type type, object? value, System.Text.Json.JsonSerializerOptions options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static System.Threading.Tasks.Task<System.Net.Http.HttpResponseMessage> PostAsJsonAsync<T>(this System.Net.Http.HttpClient client, string requestUri, T value, System.Text.Json.JsonSerializerOptions options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static System.Threading.Tasks.Task<System.Net.Http.HttpResponseMessage> PostAsJsonAsync<T>(this System.Net.Http.HttpClient client, System.Uri requestUri, T value, System.Text.Json.JsonSerializerOptions options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static System.Threading.Tasks.Task<System.Net.Http.HttpResponseMessage> PutAsJsonAsync(this System.Net.Http.HttpClient client, string requestUri, System.Type type, object? value, System.Text.Json.JsonSerializerOptions options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static System.Threading.Tasks.Task<System.Net.Http.HttpResponseMessage> PutAsJsonAsync(this System.Net.Http.HttpClient client, System.Uri requestUri, System.Type type, object? value, System.Text.Json.JsonSerializerOptions options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static System.Threading.Tasks.Task<System.Net.Http.HttpResponseMessage> PutAsJsonAsync<T>(this System.Net.Http.HttpClient client, string requestUri, T value, System.Text.Json.JsonSerializerOptions options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static System.Threading.Tasks.Task<System.Net.Http.HttpResponseMessage> PutAsJsonAsync<T>(this System.Net.Http.HttpClient client, System.Uri requestUri, T value, System.Text.Json.JsonSerializerOptions options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```

Comment:
Are developers using these APIs going to realize they need to write:
```C#
(await client.PutAsJsonAsync(...)).Dispose();
```
rather than:
```
await client.PutAsJsonAsync(...);
```
?

Reply:
Ouch, I didn't even realize that messages are disposable. That's unfortunate if we expect most people to not care about the response.

Reply:
What happens today if you don't dispose the message? I'm going to bet a lot of people have written code that doesn't dispose it.

Reply:
> What happens today if you don't dispose the message?

It depends.  Disposing the message disposes of the response content. If you used the default of HttpCompletionOption.RequestContentRead, then the impact is minimal, since the response content is just a wrapper around a fully buffered copy of the response data.  But if you used RequestHeadersRead, then the response content contains a response stream that needs to be read/disposed in order to release the underlying connection back to the connection pool... not disposing of that is a recipe for resource problems.

Reply:
It's a little bit concerning that users might get this wrong then. (though it's not different from what we had in the past)

Doesn't `HttpContent` dispose the stream when it's read?

Reply:
HttpContent only disposes the response stream when the HttpContent is disposed. SocketsHttpHandler's Stream implementations will release the underlying connection back to the pool once the Stream has fully read the output. So with SocketsHttpHandler, the main concern disappears if the consumer disposes the HttpResponseMessage, disposes the response HttpContent, disposes the response Stream, or reads the response Stream to EOF. I can't speak to whether other handlers are as diligent. And none of those are likely to happen if folks don't realize there's a possible response stream from these Post/Put operations. We could mitigate that by always buffering the response for these helpers, but that's then problematic if buffering want desired.